Ginger/GingerCore/Drivers/Appium/SeleniumAppiumDriver.cs

```diff
@@ -132,19 +141,27 @@ public enum eSwipeSide
         [UserConfiguredDescription("iOS Only | Only if 'AppInstallerPath' is empty. Bundle ID of the app under test. Useful for starting an installed app on a real device or for using other caps which require the bundle ID during test startup.")]
         public String InstalledAppBundleID { get; set; }
 
+        [UserConfigured]
+        [UserConfiguredDescription("Which automation engine to use, 'Appium' or 'UiAutomator2' or 'Espresso' for Android or 'XCUITest' for iOS or 'YouiEngine' for application built with You.i Engine. Keep empty to use default")]
+        public String AutomationName { get; set; }
+
         [UserConfigured]
         [UserConfiguredDefault("Yes")]
-        [UserConfiguredDescription("Set to 'Yes' or 'No', determine if the device window in Ginger will refresh the device screenshot after each action been performed")]
+        [UserConfiguredDescription("Set to 'Yes' or 'No', determine if the Ginger device window will be loaded with the Agent")]
+        public String LoadGingerDeviceWindow { get; set; }
+
+        [UserConfigured]
+        [UserConfiguredDefault("Yes")]
+        [UserConfiguredDescription("Set to 'Yes' or 'No', determine if the Ginger device window will refresh the device screenshot after each action been performed")]
         public String RefreshDeviceScreenShots { get; set; }
 
         [UserConfigured]
-        [UserConfiguredDefault("10")]
-        [UserConfiguredDescription("Implicit Wait for Appium Action Completion")]
-        public int ImplicitWait { get; set; }
+        [UserConfiguredDefault("120")]
+        [UserConfiguredDescription("How long (in seconds) Appium will wait for a new command from the client before assuming the client quit and ending the session")]
+        public int NewCommandTimeout { get; set; }
 
-        private static TimeSpan INIT_TIMEOUT_SEC = TimeSpan.FromSeconds(90);
-        private static TimeSpan IMPLICIT_TIMEOUT_SEC = TimeSpan.FromSeconds(10);
-        private static TimeSpan SERVER_TIMEOUT_SEC = TimeSpan.FromSeconds(600);
+        //private static TimeSpan INIT_TIMEOUT_SEC = TimeSpan.FromSeconds(90);
```

Comment:
Codacy found an issue: [Remove this commented out code.](https://app.codacy.com/manual/Ginger-Automation/Ginger/pullRequest?prid=6113827)